kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/templates/kroxylicious/KroxyliciousConfigMapTemplates.java

```diff
@@ -46,16 +53,40 @@ public static ConfigMapBuilder defaultKroxyliciousConfig(String clusterName, Str
      *
      * @param clusterName the cluster name
      * @param namespaceName the namespace name
-     * @param topicName the topic name
-     * @param config config
+     * @param testKmsFacade the test kms facade
      * @return the config map builder
      */
-    public static ConfigMapBuilder kroxyliciousRecordEncryptionConfig(String clusterName, String namespaceName, String topicName, Config config) {
+    public static ConfigMapBuilder kroxyliciousRecordEncryptionConfig(String clusterName, String namespaceName, TestKmsFacade<?, ?, ?> testKmsFacade) {
         return baseKroxyliciousConfig(namespaceName)
-                .addToData("config.yaml", getRecordEncryptionConfigMap(clusterName, topicName, config));
+                .addToData("config.yaml", getRecordEncryptionConfigMap(clusterName, testKmsFacade));
+    }
+
+    private static String buildEncryptionFilter(TestKmsFacade<?, ?, ?> testKmsFacade) {
+        return "- type: RecordEncryption"
+                + "\n  config:"
+                + "\n    kms: " + testKmsFacade.getKmsServiceClass().getSimpleName()
+                + "\n    kmsConfig:"
+                + "\n      " + getYamlKmsConfig(testKmsFacade.getKmsServiceConfig())
+                + "\n    selector: TemplateKekSelector"
+                + "\n    selectorConfig:"
+                + "\n      template: \"${topicName}\"";
```

Comment:
We assume JDK 17 so we can use a text block
```suggestion
        return """
                - type: RecordEncryption
                  config:
                    kms: %s
                    kmsConfig:
                      %s
                    selector: TemplateKekSelector
                    selectorConfig:
                      template: "${topicName}"
                """.formatted(testKmsFacade.getKmsServiceClass().getSimpleName(), getYamlKmsConfig(testKmsFacade.getKmsServiceConfig()));
```

need to double check what that does for the indentation of `-type`

Reply:
Working fine, thanks!